lib/rules/no-unused-vars.js

```diff
@@ -624,6 +640,11 @@ module.exports = {
                             if (config.varsIgnorePattern && config.varsIgnorePattern.test(def.name.name)) {
                                 continue;
                             }
+
+                            // https://github.com/eslint/eslint/issues/15611
+                            if (type === "Variable" && def.node.id.type === "ArrayPattern" && config.destructuredArrayIgnorePattern && config.destructuredArrayIgnorePattern.test(def.name.name)) {
```

Comment:
This would be a false negative on `_a`:

```js
/*eslint no-unused-vars: ["error", { "destructuredArrayIgnorePattern": "^_" }]*/

var [{ _a, foo }] = bar;
foo;
```

The rationale for this option is that it allows skipping positions in array patterns, so I think it shouldn't apply to nested patterns that are not array patterns themselves.

On the other hand, it should apply to all array patterns, so these would be false positives on `_a` and `_b`:

```js
/*eslint no-unused-vars: ["error", { "destructuredArrayIgnorePattern": "^_" }]*/

var { x: [_a, foo] } = bar;
foo;

function baz([_b, foo]) {
    foo;
}
baz();

```

Reply:
Updated PR to cover these cases too and added corresponding test cases.